src/drive/storage/index.ts

```diff
@@ -0,0 +1,381 @@
+
+      pendingFolders.shift();
+
+      // * Renames current folder files
+      for (const file of files) {
+        const relativePath = `${currentFolder.destinationPath}/${this.fileDisplayName(file)}`;
+        const hashedPath = hashPath(relativePath);
+        await this.updateFileReference(file.fileId, bucketId, hashedPath);
+      }
+
+      // * Adds current folder folders to pending
+      pendingFolders.push(
+        ...folders.map((folderData) => ({
+          destinationPath: `${currentFolder.destinationPath}/${folderData.name}`,
+          folderId: folderData.id,
+        })),
+      );
+    }
+  }
+
+  /**
+   * Updates the stored reference of a file in the centralized persistence
+   * @param fileId
+   * @param bucketId
+   * @param hashedPath
+   * @private
+   */
+  private updateFileReference(
+    fileId: string,
+    bucketId: string,
+    hashedPath: string,
+  ): Promise<void> {
+    return this.axios
+      .post(`${this.apiUrl}/api/storage/rename-file-in-network`, {
+        fileId: fileId,
+        bucketId: bucketId,
+        relativePath: hashedPath,
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+  }
+
+
+  /**
+   * Creates a new file entry
+   * @param fileEntry
+   */
+  public createFileEntry(fileEntry: FileEntry): Promise<unknown> {
+    return this.axios
+      .post(`${this.apiUrl}/api/storage/file`, {
+        fileId: fileEntry.id,
+        type: fileEntry.type,
+        bucket: fileEntry.bucket,
+        size: fileEntry.size,
+        folder_id: fileEntry.folder_id,
+        name: fileEntry.name,
+        encrypt_version: fileEntry.encrypt_version,
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      })
+      .catch(error => {
+        throw new Error(extractAxiosErrorMessage(error));
+      });
+  }
+
+  /**
+   * Updates the details of a file entry
+   * @param payload
+   * @param hashPath
+   */
+  public updateFile(payload: UpdateFilePayload, hashPath: HashPath) {
+    const hashedPath = hashPath(payload.destinationPath);
+    return this.axios
+      .post(`${this.apiUrl}/api/storage/file/${payload.fileId}/meta`, {
+        metadata: payload.metadata,
+        bucketId: payload.bucketId,
+        relativePath: hashedPath,
+      }, {
+        headers: this.headers()
+      });
+  }
+
+  /**
+   * Deletes a specific file entry
+   * @param payload
+   */
+  public deleteFile(payload: DeleteFilePayload) {
+    return this.axios
+      .delete(`${this.apiUrl}/api/storage/folder/${payload.folderId}/file/${payload.fileId}`, {
+        headers: this.headers()
+      });
+  }
+
+  /**
+   * Updates the persisted path of a file entry
+   * @param payload
+   * @param hashPath
+   */
+  public moveFile(payload: MoveFilePayload, hashPath: HashPath): Promise<MoveFileResponse> {
+    const hashedPath = hashPath(payload.destinationPath);
```

Comment:
now this should use a UUID as making the SDK as compatible as possible, the `relativePath` is going to be passed as a parameter instead of generating a UUID here, as React Native has incompatibilities with that library